openshift-kubecost-install.md

```diff
@@ -0,0 +1,513 @@
+          relabel_configs:
+            - replacement: kubernetes.default.svc:443
+              target_label: __address__
+            - regex: (.+)
+              replacement: /api/v1/nodes/$1/proxy/metrics
+              source_labels:
+                - __meta_kubernetes_node_name
+              target_label: __metrics_path__
+          scheme: https
+          tls_config:
+              ca_file: /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
+              insecure_skip_verify: false
+              server_name: kubernetes
+  
+EOF
+(export NAMESPACE=kubecost && kubectl apply -n $NAMESPACE -f -)
+
+MANIFEST_URL=https://raw.githubusercontent.com/kubecost/openshift-helm-chart/main/grafana-agent-config/agent-bare.yaml NAMESPACE=kubecost /bin/sh -c "$(curl -fsSL https://raw.githubusercontent.com/grafana/agent/v0.24.2/production/kubernetes/install-bare.sh)" | kubectl apply -f -
+```
+</details>
+
+To learn more about how to install and config Grafana agent as well as additional scrape configuration, please refer to [Grafana Agent for Kubernetes](https://grafana.com/docs/grafana-cloud/kubernetes/agent-k8s/k8s_agent_metrics/) section of the Grafana Cloud documentation. Or you can check Kubecost Prometheus scrape config at this [Github repository](https://github.com/kubecost/cost-analyzer-helm-chart/blob/ebe7e088debecd23f90e6dd75b425828901a246c/cost-analyzer/charts/prometheus/values.yaml#L1152)
+
+**3. Verify if grafana-agent is scraping data successfully:**
+
+`kubectl -n kubecost logs grafana-agent-0`
+
+**4. Create dbsecret to allow Kubecots to query the metrics from Grafana Cloud Prometheus:**
+
+- Create two files in your working directory, called `USERNAME` and `PASSWORD` respectively
+  
+```Bash
+export PASSWORD=<REPLACE-WITH-GRAFANA-PROM-REMOTE-WRITE-API-KEY>
+export USERNAME=<REPLACE-WITH-GRAFANA-PROM-REMOTE-WRITE-USERNAME>
+printf "${PASSWORD}" > PASSWORD
+printf "${USERNAME}" > USERNAME
+```
+- Verify that you can run query against your Grafana Cloud Prometheus query endpoint with your API key (Optional):
+
+```Bash
+cred="$( echo $NAME:$PASSWORD | base64 )"; curl -H "Authorization: Basic $cred" https://<REPLACE-WITH-GRAFANA-PROM-QUERY-ENDPOINT>/api/v1/query?query=up
+```
+- Create K8s secret name dbsecret:
+
+```Bash
+kubectl create secret generic dbsecret \
+    --namespace kubecost \
+    --from-file=USERNAME \
+    --from-file=PASSWORD
+```
+- Verify if the credentials appears correctly - Optional (Any trailing space or new line etc ...)
+
+```Bash
+kubectl -n kubecost get secret dbsecret -o json | jq '.data | map_values(@base64d)'
+```
+**5. (optional): Configure Kubecost recording rules for Grafana Cloud using cortextool**
+
+To set up recording rules in Grafana Cloud, download the [cortextool CLI utility](https://github.com/grafana/cortex-tools). While they are optional, they offer improved performance.
+
+After installing the tool, create a file called \_kubecost\_rules.yaml\_ with the following command:
```

Comment:
Not sure where `\_` is from, but based on the code sample below, it looks like it should be `kubecost-rules.yaml`
```suggestion
After installing the tool, create a file called `kubecost-rules.yaml` with the following command:
```